### ISSUE TITLE
[BUG] Multi module deploy breaks when including deploy hook in a module

### ISSUE THREAD

**Issue by DennisVis** [GoogleContainerTools/skaffold#6627]
### Expected behavior
Skaffold is able to run multiple modules whether they contain a deploy hook or not.

### Actual behavior
When one of the modules given to Skaffold to run contains a kustomize deploy hook, only that module is deployed, the others are only built.

### Information
- Skaffold version: v1.32.0
- Operating system: MacOS 11.4
- Installed via: brew

### Steps to reproduce the behavior
1. Create a multi-module Skaffold project
2. Add a deploy hook to one of the modules
3. Run `skaffold run` supplying all modules using the `-m` flag

When removing the deploy hook everything runs as expected.

**Comment by briandealwis** [comment]
I'm unable to reproduce @DennisVis.  Here are my steps:

1. `cd examples/multi-config-microservices`
2. patch `leeroy-web/skaffold.yaml` to add before-and-after deploy hooks:
    ```
    --- examples/multi-config-microservices/leeroy-web/skaffold.yaml
    +++ examples/multi-config-microservices/leeroy-web/skaffold.yaml
    @@ -14,6 +14,13 @@ deploy:
       kubectl:
         manifests:
         - kubernetes/*
    +    hooks:
    +      before:
    +      - host:
    +          command: [echo, "before-deploy-hook"]
    +      after:
    +      - host:
    +          command: [echo, "after-deploy-hook"]
     portForward:
     - resourceType: deployment
       resourceName: leeroy-web
    ```
3. `skaffold run -m app-config,web-config --tail`

I see both the `app-config` and `web-config` modules deploy.
```
Starting deploy...
 - service/leeroy-app created
 - deployment.apps/leeroy-app created
Starting pre-deploy hooks...
before-deploy-hook
Completed pre-deploy hooks
 - deployment.apps/leeroy-web created
Waiting for deployments to stabilize...
 - deployment/leeroy-app is ready. [1/2 deployment(s) still pending]
 - deployment/leeroy-web is ready.
Deployments stabilized in 2.222 seconds
Starting post-deploy hooks...
after-deploy-hook
Completed post-deploy hooks
Waiting for deployments to stabilize...
Deployments stabilized in 7.714302ms
Press Ctrl+C to exit
[leeroy-app] 2021/10/01 14:20:58 leeroy app server ready
[leeroy-web] 2021/10/01 14:20:59 leeroy web server ready
```

Could you please provide a small test case for reproduction?

**Comment by DennisVis** [comment]
I was having trouble reproducing in a new repro project as well. I can reproduce in my real project consistently though, but not in all modules. When commenting out the command or moving the hooks to the build instead of deploy everything works as expected (the latter being my current workaround).

Will try to get you a repro project next week, based on the config of the modules that are failing for me.

**Comment by gsquared94** [comment]
@DennisVis if any of the deploy hooks fail then skaffold doesn't proceed with deploying the remaining modules. Is it possible that the hook command is erroring out? Can you try running with `-v DEBUG` flag?

**Comment by DennisVis** [comment]
@gsquared94 I considered that possibility as well and to test it out reduced the hooks to simple echo commands, even adding a || true to make sure they succeeded. I still noticed the same behaviour. Moving the actual hook logic to a build hook instead of a deploy hook also fixed the issue for me. Therefore I suspect the issue is not with the actual commands of the hooks.

I haven't ran them with the DEBUG flag enabled yet, will do that now to see if something stands out.

**Comment by DennisVis** [comment]
I'm not able to reproduce anymore. I had made some changes since reporting the issue to make it work for me and it seems one of the changes I made resolved the issue.

I can only guess as to the cause of the issue now but I suspect it either was some kind of configuration error on my side or otherwise a bug with a very specific configuration setup that will likely be very rare.

I think this ticket can be closed for now. If I do happen to bump into it again I'll be sure to immediately isolate it in a repro and create a new ticket.
